File: pc-editor/src/main/java/com/auto_persona/pc/model/GameData.java
package com.auto_persona.pc.model;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class GameData {
    public String version = "1.0.0";
    public long timestamp;
    public PlayerInfo playerInfo = new PlayerInfo();
    public String gender = "brother";
    public CharacterStats characterStats = new CharacterStats();
    public CharacterSystemData characterSystemData = new CharacterSystemData();
    public PlayerProgress playerProgress = new PlayerProgress();
    public QuestSystem questSystem = new QuestSystem();
    public GlobalLevelSystem globalLevelSystem = new GlobalLevelSystem();
    public TimeSystem timeSystem = new TimeSystem();
    public Inventory inventory = new Inventory();
    public GameSettings settings = new GameSettings();
    public DateHistory dateHistory = new DateHistory();
    public TravelSystem travelSystem = new TravelSystem();
    public ShopSystem shopSystem = new ShopSystem();
    public String currentOutfit;
    public int kemonomimiReminderCount;

    // ---- Sub-classes ----

    public static class PlayerInfo {
        public String name = "";
        public String birthday = "";
        public String identity = "";
        public boolean hasCollected;
    }

    public static class CharacterStats {
        public int affection;
        public int trust;
    }

    public static class CharacterSystemData {
        public CharacterStats stats = new CharacterStats();
        public CharacterInfo character = new CharacterInfo();
        public boolean diaryLeakTriggered;
        public boolean yandereEventTriggered;
        public boolean giftPreferenceEventTriggered;
        public boolean affection30EventTriggered;
        public boolean affection70EventTriggered;
        public int affectionCap = 1000;
        public int trustMin;
        public int trustMax = 100;
        public int kemonomimiShyness;
        public boolean kemonomimiCatForm;
        public String currentPersonaPackId;
        public boolean diaryLeakRecovered;
        public boolean yandereRecovered;
        public boolean outfit19Unlocked;
    }

    public static class CharacterInfo {
        public String name = "YUKI";
        public String personality = "";
        public List<String> favoriteGifts = new ArrayList<>();
        public List<String> specialEvents = new ArrayList<>();
        public String currentMood = "normal";
    }

    public static class PlayerProgress {
        public int coins;
    }

    public static class QuestSystem {
        public int unlockedLevel;
    }

    public static class GlobalLevelSystem {
        public int globalLevel = 1;
        public int globalExp;
    }

    public static class TimeSystem {
        public int currentDay = 1;
        public int dailyActions = 99999;
        public int actionsUsed;
        public List<String> specialEvents = new ArrayList<>();
        public String gameStartDate = "";
    }

    public static class Inventory {
        public List<InventoryItem> items = new ArrayList<>();
        public Equipment equipment = new Equipment();
    }

    public static class InventoryItem {
        public String id = "";
        public String name = "";
        public int quantity;
        public String type = "";
    }

    public static class Equipment {
        public String weapon;
        public String armor;
        public String accessory;
    }

    public static class GameSettings {
        public double musicVolume = 0.7;
        public double soundVolume = 0.8;
        public boolean autoSave = true;
    }

    public static class DateHistory {
        public int custom, cafe, park, restaurant, beach, nightmarket, nightpath;
    }

    public static class TravelSystem {
        public boolean hasShownNightDialogue;
        public Map<String, String> travelHistory = new HashMap<>();
        public List<String> unlockedDestinations = new ArrayList<>();
        public List<String> currentSpots = new ArrayList<>();
        public String travelState, currentDestination, destinationCityName, currentCity;
        public boolean isTravelMode;
        public String travelStartTime, travelEndTime;
    }

    public static class ShopSystem {
        public OutfitPurchases purchases = new OutfitPurchases();
        public GiftInventory giftInventory = new GiftInventory();
        public Map<String, Integer> travelInventory = new HashMap<>();
        public String currentTab = "travel";
    }

    public static class OutfitPurchases {
        public boolean outfit1, outfit2, outfit3, outfit4, outfit5,
               outfit6, outfit7, outfit8, outfit9, outfit10,
               outfit11, outfit12, outfit13, outfit14, outfit15,
               outfit16, outfit17, outfit18, outfit19, outfit20;
    }

    public static class GiftInventory {
        public int gift1, gift2, gift3, gift4, gift5, gift6,
                   gift7, gift8, gift9, gift10, gift11, gift12;
    }
}

File: app/src/main/java/com/example/auto_persona/data/model/GameData.kt
package com.example.auto_persona.data.model

import kotlinx.serialization.Serializable

@Serializable
data class GameData(
    val version: String = "1.0.0",
    val timestamp: Long = 0L,
    val playerInfo: PlayerInfo = PlayerInfo(),
    val gender: String = "brother",
    val characterStats: CharacterStats = CharacterStats(),
    val characterSystemData: CharacterSystemData = CharacterSystemData(),
    val playerProgress: PlayerProgress = PlayerProgress(),
    val questSystem: QuestSystem = QuestSystem(),
    val globalLevelSystem: GlobalLevelSystem = GlobalLevelSystem(),
    val timeSystem: TimeSystem = TimeSystem(),
    val inventory: Inventory = Inventory(),
    val settings: GameSettings = GameSettings(),
    val dateHistory: DateHistory = DateHistory(),
    val travelSystem: TravelSystem = TravelSystem(),
    val shopSystem: ShopSystem = ShopSystem(),
    val currentOutfit: String? = null,
    val kemonomimiReminderCount: Int = 0
)

@Serializable
data class PlayerInfo(
    val name: String = "",
    val birthday: String = "",
    val identity: String = "",
    val hasCollected: Boolean = false
)

@Serializable
data class CharacterStats(
    val affection: Int = 0,
    val trust: Int = 0
)

@Serializable
data class CharacterSystemData(
    val stats: CharacterStats = CharacterStats(),
    val character: CharacterInfo = CharacterInfo(),
    val diaryLeakTriggered: Boolean = false,
    val yandereEventTriggered: Boolean = false,
    val giftPreferenceEventTriggered: Boolean = false,
    val affection30EventTriggered: Boolean = false,
    val affection70EventTriggered: Boolean = false,
    val affectionCap: Int = 1000,
    val trustMin: Int = 0,
    val trustMax: Int = 100,
    val kemonomimiShyness: Int = 0,
    val kemonomimiCatForm: Boolean = false,
    val currentPersonaPackId: String? = null,
    val diaryLeakRecovered: Boolean = false,
    val yandereRecovered: Boolean = false,
    val outfit19Unlocked: Boolean = false
)

@Serializable
data class CharacterInfo(
    val name: String = "YUKI",
    val personality: String = "",
    val favoriteGifts: List<String> = emptyList(),
    val specialEvents: List<String> = emptyList(),
    val currentMood: String = "normal"
)

@Serializable
data class PlayerProgress(
    val coins: Int = 0
)

@Serializable
data class QuestSystem(
    val unlockedLevel: Int = 0
)

@Serializable
data class GlobalLevelSystem(
    val globalLevel: Int = 1,
    val globalExp: Int = 0
)

@Serializable
data class TimeSystem(
    val currentDay: Int = 1,
    val dailyActions: Int = 99999,
    val actionsUsed: Int = 0,
    val specialEvents: List<String> = emptyList(),
    val gameStartDate: String = ""
)

@Serializable
data class Inventory(
    val items: List<InventoryItem> = emptyList(),
    val equipment: Equipment = Equipment()
)

@Serializable
data class InventoryItem(
    val id: String = "",
    val name: String = "",
    val quantity: Int = 0,
    val type: String = ""
)

@Serializable
data class Equipment(
    val weapon: String? = null,
    val armor: String? = null,
    val accessory: String? = null
)

@Serializable
data class GameSettings(
    val musicVolume: Double = 0.7,
    val soundVolume: Double = 0.8,
    val autoSave: Boolean = true
)

@Serializable
data class DateHistory(
    val custom: Int = 0,
    val cafe: Int = 0,
    val park: Int = 0,
    val restaurant: Int = 0,
    val beach: Int = 0,
    val nightmarket: Int = 0,
    val nightpath: Int = 0
)

@Serializable
data class TravelSystem(
    val hasShownNightDialogue: Boolean = false,
    val travelHistory: Map<String, String> = emptyMap(),
    val unlockedDestinations: List<String> = emptyList(),
    val currentSpots: List<String> = emptyList(),
    val travelState: String? = null,
    val currentDestination: String? = null,
    val destinationCityName: String? = null,
    val currentCity: String? = null,
    val isTravelMode: Boolean = false,
    val travelStartTime: String? = null,
    val travelEndTime: String? = null
)

@Serializable
data class ShopSystem(
    val purchases: OutfitPurchases = OutfitPurchases(),
    val giftInventory: GiftInventory = GiftInventory(),
    val travelInventory: Map<String, Int> = emptyMap(),
    val currentTab: String = "travel"
)

@Serializable
data class OutfitPurchases(
    val outfit1: Boolean = false, val outfit2: Boolean = false,
    val outfit3: Boolean = false, val outfit4: Boolean = false,
    val outfit5: Boolean = false, val outfit6: Boolean = false,
    val outfit7: Boolean = false, val outfit8: Boolean = false,
    val outfit9: Boolean = false, val outfit10: Boolean = false,
    val outfit11: Boolean = false, val outfit12: Boolean = false,
    val outfit13: Boolean = false, val outfit14: Boolean = false,
    val outfit15: Boolean = false, val outfit16: Boolean = false,
    val outfit17: Boolean = false, val outfit18: Boolean = false,
    val outfit19: Boolean = false, val outfit20: Boolean = false
)

@Serializable
data class GiftInventory(
    val gift1: Int = 0, val gift2: Int = 0, val gift3: Int = 0,
    val gift4: Int = 0, val gift5: Int = 0, val gift6: Int = 0,
    val gift7: Int = 0, val gift8: Int = 0, val gift9: Int = 0,
    val gift10: Int = 0, val gift11: Int = 0, val gift12: Int = 0
)
